src/Button/Button.component.tsx

```diff
@@ -3,113 +3,540 @@ import * as React from 'react';
-    primary: css`
-        border: solid thin ${colors.savvyCyan.base};
-        background-color: ${colors.savvyCyan.base};
-        color: ${colors.white.base};
+const themeDefaults = {
+    primary: colors.savvyCyan,
+    outline: {
+        base: colors.savvyCyan.dark,
+        light: colors.savvyCyan.base,
+        dark: colors.savvyCyan.dark,
+    },
+    minimal: {
+        base: colors.savvyCyan.dark,
+        light: colors.savvyCyan.base,
+        dark: colors.savvyCyan.dark,
+    },
+};
 
-        &:hover {
-            background-color: ${darken(0.05, colors.savvyCyan.base)};
-        }
-    `,
-    secondary: css`
-        border: solid thin ${colors.savvyCyan.light};
-        background-color: ${colors.savvyCyan.light};
-        color: ${colors.white.base};
-
-        &:hover {
-            background-color: ${darken(0.1, colors.savvyCyan.light)};
-        }
-    `,
-    text: css`
-        border: solid thin transparent;
-        background-color: transparent;
-        color: ${colors.savvyCyan.base};
-
-        &:hover {
-            color: ${darken(0.25, colors.savvyCyan.base)};
-        }
-    `,
+const reverseDefaults = {
+    primary: {
+        base: colors.charcoal.light,
+        light: colors.charcoal.light,
+        dark: colors.charcoal.dark,
+    },
+    outline: {
+        base: colors.white.base,
+        light: colors.white.base,
+        dark: colors.charcoal.light,
+    },
+    minimal: colors.white,
 };
 
-const ButtonSizeStyles: ButtonStylesGroup = {
-    small: css`
-        padding: ${sizes.padding.xs};
-        font-size: 0.75rem;
-    `,
-    medium: css`
-        padding: ${sizes.padding.sm} ${sizes.padding.md};
-        font-size: 1rem;
-    `,
-    large: css`
-        padding: ${sizes.padding.md};
-        font-size: 1.5rem;
-    `,
+const dimensions = {
+    xs: {
+        width: 4,
+        height: 2,
+        padding: 0.5,
+        circularPadding: 1,
+        fontSize: 0.75,
+    },
+    sm: {
+        width: 5,
+        height: 2.5,
+        padding: 1,
+        circularPadding: 1.5,
+        fontSize: 0.875,
+    },
+    md: {
+        width: 12.5,
+        height: 3,
+        padding: 1.5,
+        circularPadding: 2,
+        fontSize: 1,
+    },
+    lg: {
+        width: 12.5,
+        height: 3.5,
+        padding: 2,
+        circularPadding: 2.5,
+        fontSize: 1,
+    },
 };
 
-const StyledButton = styled.button`
+interface ButtonStyles {
+    base: FlattenSimpleInterpolation;
+    disabled: FlattenSimpleInterpolation;
+    hover: FlattenSimpleInterpolation;
+    active?: FlattenSimpleInterpolation;
+    focus?: FlattenSimpleInterpolation;
+    focusOutline?: FlattenSimpleInterpolation;
+}
+
+interface ButtonStylesGroup {
+    [key: string]: ButtonStyles;
+}
+
+const ButtonColorStyles = ({
+    colorTheme,
+    reverse,
+}: {
+    colorTheme: Theme;
+    reverse?: boolean;
+}): ButtonStylesGroup => {
+    const { base, dark, light } = colorTheme;
+
+    if (reverse) {
```

Comment:
I feel like lines 137 - 281 can largely be abstracted into separate files, making this particular file ~150 lines shorter and easier to read, unless I'm missing why it needs to here.

Reply:
As I was brutally shot down, never-mind this.